src/NServiceBus.Core/Pipeline/MutateInstanceMessage/MutateOutgoingMessageBehavior.cs

```diff
@@ -1,32 +1,28 @@
 ﻿namespace NServiceBus
 {
     using System;
-    using System.Collections.Generic;
     using System.Threading.Tasks;
     using MessageMutator;
     using NServiceBus.Pipeline.OutgoingPipeline;
+    using NServiceBus.Transports;
+    using NServiceBus.Unicast.Messages;
     using Pipeline;
-    using Pipeline.Contexts;
 
     class MutateOutgoingMessageBehavior : Behavior<OutgoingLogicalMessageContext>
     {
         public override async Task Invoke(OutgoingLogicalMessageContext context, Func<Task> next)
         {
-            InvokeHandlerContext incomingState;
-            context.TryGetRootContext(out incomingState);
+            LogicalMessage incomingMessage;
+            context.TryGet(out incomingMessage);
+
+            IncomingMessage incomingPhysicalMessage;
+            context.TryGet(out incomingPhysicalMessage);
```

Comment:
incomingMessage : LogicalMessage
incomingPhysicalMessage : IncomingMessage

the naming seems a bit confusing. How about:
incomingPhysicalMessage : IncomingMessage
incomingLogicalMessage : LogicalMessage